packages/rrweb/src/replay/index.ts

```diff
@@ -1835,17 +1835,22 @@ export class Replayer {
                   const newSn = mirror.getMeta(
                     target as Node & RRNode,
                   ) as serializedElementNodeWithId;
-                  Object.assign(
-                    newSn.attributes,
-                    mutation.attributes as attributes,
+                  const newNode = buildNodeWithSN(
+                    {
+                      ...newSn,
+                      attributes: {
+                        ...newSn.attributes,
+                        ...(mutation.attributes as attributes),
+                      },
+                    },
+                    {
+                      doc: target.ownerDocument as Document, // can be Document or RRDocument
+                      mirror: mirror as Mirror,
+                      skipChild: true,
+                      hackCss: true,
+                      cache: this.cache,
+                    },
                   );
-                  const newNode = buildNodeWithSN(newSn, {
```

Comment:
If you look above at the `Object.assign` call, you'll see that we mutate the meta object reference from the mirror and pass it to `buildNodeWithSN()`. Looking [inside of `buildNodeWithSN()`](https://github.com/rrweb-io/rrweb/blob/master/packages/rrweb-snapshot/src/rebuild.ts#L432-L438) you can see that it ends up comparing the same object reference and returning the node from the mirror instead of building a new node.

```javascript
if (mirror.has(n.id)) {
    const nodeInMirror = mirror.getNode(n.id)!;
    const meta = mirror.getMeta(nodeInMirror)!;
    if (isNodeMetaEqual(meta, n)) return mirror.getNode(n.id);
  }
```

This meant that remote stylesheets were always loaded remotely (even if you inline) and could be blocked by CORS, making the inlined CSS rules unused.